File: core/src/main/java/de/qabel/core/config/Account.java
package de.qabel.core.config;

/**
 * https://github.com/Qabel/qabel-doc/wiki/Qabel-Client-Configuration#account
 */
public class Account extends SyncSettingItem {
    private static final long serialVersionUID = -6902585443982494539L;
    /**
     * Provider of the account
     * Field name in serialized json: "provider"
     */
    private String provider;
    /**
     * User of the account
     * Field name in serialized json: "user"
     */
    private String user;
    /**
     * Authentication of the account
     * Field name in serialized json: "auth"
     */
    private String auth;
    /**
     * AuthToken of the account
     * May expire but should be used until proven to be expired
     */
    private String token;

    /**
     * Creates an instance of Account
     *
     * @param provider Provider of the account.
     * @param user     User of the account
     * @param auth     Authentication of the account
     */
    public Account(String provider, String user, String auth) {
        this(provider, user);
        setAuth(auth);
    }

    /**
     * Creates an instance of Account
     *
     * @param provider Provider of the account.
     * @param user     User of the account
     */
    public Account(String provider, String user) {
        setProvider(provider);
        setUser(user);
    }

    public void setProvider(String value) {
        provider = value;
    }

    public String getProvider() {
        return provider;
    }

    public void setUser(String value) {
        user = value;
    }

    public String getUser() {
        return user;
    }

    public void setAuth(String value) {
        auth = value;
    }

    public String getAuth() {
        return auth;
    }

    public String getToken() {
        return token;
    }

    public void setToken(String token) {
        this.token = token;
    }

    @Override
    public int hashCode() {
        final int prime = 31;
        int result;

        result = super.hashCode();

        result = prime * result + (provider == null ? 0 : provider.hashCode());
        result = prime * result + (user == null ? 0 : user.hashCode());
        return result;
    }

    @Override
    public boolean equals(Object obj) {
        if (!super.equals(obj)) {
            return false;
        }

        if (this == obj) {
            return true;
        }
        if (getClass() != obj.getClass()) {
            return false;
        }

        Account other = (Account) obj;
        if (auth == null) {
            if (other.auth != null) {
                return false;
            }
        } else if (!auth.equals(other.auth)) {
            return false;
        }
        if (provider == null) {
            if (other.provider != null) {
                return false;
            }
        } else if (!provider.equals(other.provider)) {
            return false;
        }
        if (user == null) {
            return other.user == null;
        }
        return user.equals(other.user);
    }
}

File: core/src/main/java/de/qabel/core/config/Account.kt
package de.qabel.core.config

/**
 * https://github.com/Qabel/qabel-doc/wiki/Qabel-Client-Configuration#account
 */
class Account
/**
 * Creates an instance of Account

 * @param provider Provider of the account.
 * *
 * @param user     User of the account
 * *
 * @param auth     Authentication of the account
 */
(provider: String, user: String, auth: String) : SyncSettingItem() {
    /**
     * Provider of the account
     * Field name in serialized json: "provider"
     */
    var provider: String? = null
    /**
     * User of the account
     * Field name in serialized json: "user"
     */
    var user: String? = null
    /**
     * Authentication of the account
     * Field name in serialized json: "auth"
     */
    var auth: String? = null

    init {
        this.provider = provider
        this.user = user
        this.auth = auth
    }

    override fun hashCode(): Int {
        val prime = 31
        var result = 1

        result = super.hashCode()

        result = prime * result + if (auth == null) 0 else auth!!.hashCode()
        result = prime * result + if (provider == null) 0 else provider!!.hashCode()
        result = prime * result + if (user == null) 0 else user!!.hashCode()
        return result
    }

    override fun equals(obj: Any?): Boolean {
        if (super.equals(obj) == false) {
            return false
        }

        if (this === obj) {
            return true
        }
        if (obj == null) {
            return false
        }
        if (javaClass != obj.javaClass) {
            return false
        }

        val other = obj as Account?
        if (auth == null) {
            if (other!!.auth != null) {
                return false
            }
        } else if (auth != other!!.auth) {
            return false
        }
        if (provider == null) {
            if (other.provider != null) {
                return false
            }
        } else if (provider != other.provider) {
            return false
        }
        if (user == null) {
            if (other.user != null) {
                return false
            }
        } else if (user != other.user) {
            return false
        }
        return true
    }

    companion object {
        private val serialVersionUID = -6902585443982494539L
    }
}
